### ISSUE TITLE
Command parameter aliases

### ISSUE THREAD

**Issue by Hwiggy** [TheSourceCodeLLC/Source#11]
Not certain how to explain this, so theorize the following:

!dice <rolls> <sides> -> Roll an x sided dice y times.

!1d6 -> roll a 6 sided dice 1 time.
!4d10 -> roll a 10 sided dice 4 times.

**Comment by TheForbiddenAi** [comment]
Rather than baking the arguments into the command itself could just be like !roll (# of dice) (# of sides)

**Comment by Hwiggy** [comment]
That defeats the purpose of the command parameter alias.
The idea is that some shorthand command can be expanded to run a full command with arguments derived by from alias.

**Comment by Hwiggy** [comment]
API Showcase; relevant files:
https://github.com/TheSourceCodeLLC/Source/blob/feature/param_alias/Extensions/Dice/src/main/kotlin/net/sourcebot/module/dice/RollCommand.kt

https://github.com/TheSourceCodeLLC/Source/blob/feature/param_alias/API/src/main/kotlin/net/sourcebot/api/command/CommandHandler.kt#L37

**Comment by Dhruvin-Purohit** [comment]
> Rather than baking the arguments into the command itself could just be like !roll (# of dice) (# of sides)
> 
> I don't see the point in baking the arguments into the command

What I would recommend is like if I do !roll it simply takes the default no. of sides to 6 and no. of outcomes to 1

and if I do !roll 9 it'd change the number of sides to 9 and still keep the default number of outcomes i.e. 1

so !roll 9 8 would roll a dice with 9 sides and give 8 outcomes 

and if !roll 13 18 is done it just returns an error message saying you can't have more outcomes than sides

in simple form:
```
!roll [ no. of sides (default: 6) ] [ no. of outcomes (default: 1) ]
```

**Comment by Hwiggy** [comment]
You are missing the point of command parameter aliases.
This is not a discussion on the syntax or semantics of a roll command, it was merely used as an example.

**Comment by Hwiggy** [comment]
Closed by commit ea8f664
